src/precompiled.hpp

```diff
@@ -44,6 +44,9 @@
 
 #define __STDC_LIMIT_MACROS
 
+// This must be defined before including <stdio.h> to get snprintf_s.
+#define __STDC_WANT_LIB_EXT1__ 1
```

Comment:
These will not be available everywhere, especially on old compilers and systems, so they cannot be used. Again, please just check the return value in the 2 cases where it's actually needed, and ignore the rest.